### PR TITLE
Change DPC++ compiler to clang++ in building scripts.

### PR DESCRIPTION
**Pull request by KalyanovD**
# Description
Switch dpcpp to clang++

## Changes proposed in this pull request:
- building host part with clang++
- building oneapi part with clang++ -fsycl

### CODE REVIEW

**Comment by KalyanovD** [comment]
http://intel-ci.intel.com/eb6d584f-c545-f112-9681-8cdcd4b723a1

**Comment by KalyanovD** [comment]
http://intel-ci.intel.com/eb707dfe-ae23-f16e-aadb-8cdcd4b723a1

**Comment by PetrovKP** [comment]
@Mergifyio rebase

**Review by PetrovKP — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -274,7 +274,9 @@ def getpyexts():
     ela = []
 
     if using_intel and IS_WIN:
-        if os.environ.get('cc', '') == "dpcpp":
+        if os.environ.get('cc', '') == "dpcpp" or \
+                os.environ.get('cc', '') == "clang++" or \
+                os.environ.get('cc', '') == "clang-cl":
```
Comment (ivan-samsonov) [inline setup.py:279]: Probably, will be easier:
```suggestion
        if os.environ.get('cc', '') in ("dpcpp", "clang++", "clang-cl"):
```

File: setup.py [hunk 2]
```diff
@@ -54,7 +54,7 @@
         # noinspection PyUnresolvedReferences
         import dpcppcompiler
         sys.modules["distutils.dpcppcompiler"] = sys.modules["dpcppcompiler"]
-        distutils.ccompiler.compiler_class["dpcpp"] = (
+        distutils.ccompiler.compiler_class["clang-cl"] = (
             "dpcppcompiler", "DPCPPCompiler", "Support of DPCPP compiler"
```
Comment (ivan-samsonov) [inline setup.py:58]: May be we should rename DPCPP to clang in compiler definition as well?

**Review by ivan-samsonov — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (KalyanovD) [reply setup.py]: Good point, thanks!

File: setup.py (hunk 2 above)
Comment (KalyanovD) [reply setup.py:58]: I don't think it is necessary, because we propose this changes to be able to build dpcpp functionality as well.

**Review by Pahandrovich — APPROVED** [review]

**Pull request merged**